tstore/tslong/dask.py

```diff
@@ -1,16 +1,125 @@
-        """Open a TStore file structure as a TSLongDask wrapper around a Dask long dataframe."""
+    def from_tstore(
+        base_dir,
+        ts_variables=None,
+        start_time=None,
+        end_time=None,
+        tstore_ids=None,
+        columns=None,
+        filesystem=None,
+        use_threads=True,
+    ) -> "TSLongDask":
+        """Open a TStore file structure as a TSLongDask wrapper around a Pandas long dataframe."""
+        # Read exploiting pyarrow
+        tslong_pyarrow = TSLongPyArrow.from_tstore(
+            base_dir,
+            ts_variables=ts_variables,
+            start_time=start_time,
+            end_time=end_time,
+            tstore_ids=tstore_ids,
+            columns=columns,
+            filesystem=filesystem,
+            use_threads=use_threads,
+        )
+
+        # Conversion to pandas
+        return tslong_pyarrow.change_backend(new_backend="dask")
+
+    def to_tsdf(self) -> "TSDFDask":
+        """Convert the wrapper into a TSDFDask object."""
+        from tstore.tsdf.dask import TSDFDask
+
+        tstore_ids = self._get_tstore_ids()
+        ts_arrays = self._get_ts_arrays()
+        pd_series = {ts_variable: pd.Series(ts_array, index=tstore_ids) for ts_variable, ts_array in ts_arrays.items()}
+        static_values = self._get_static_values()
+        data = {**pd_series, **static_values, self._tstore_id_var: tstore_ids}
+
+        df = pd.DataFrame(data)
+        return TSDFDask(
+            df,
+            id_var=self._tstore_id_var,
+        )
+
+    def _get_tstore_ids(self) -> list:
+        """Retrieve the list of tstore ids."""
+        return sorted(self._obj[self._tstore_id_var].unique())
+
+    def _get_ts_arrays(self) -> dict[str, TSArray]:
+        """Create a dictionary of TSArrays."""
+        return {ts_variable: self._get_ts_array(variables) for ts_variable, variables in self._tstore_ts_vars.items()}
+
+    def _get_ts_array(self, variables: list[str]) -> TSArray:
+        """Create a TSArray for a set of variables."""
+        tstore_ids = self._get_tstore_ids()
+        ts_list = [self._get_ts(tstore_id, variables) for tstore_id in tstore_ids]
+        return TSArray(ts_list)
+
+    def _get_ts(self, tstore_id: str, variables: list[str]) -> TS:
+        """Create a TS object for a given tstore_id and a set of variables."""
+        df = self._obj
+        df = df[df[self._tstore_id_var] == tstore_id]
+        df = df[variables]
+        return TS(df)
+
+    def _get_static_values(self) -> dict[str, list]:
+        """Retrieve the static values."""
+        df = self._obj
+        tstore_ids = self._get_tstore_ids()
+        static_values = {}
+
+        for static_var in self._tstore_static_vars:
+            var_static_values = []
+            for tstore_id in tstore_ids:
+                values = df[df[self._tstore_id_var] == tstore_id][static_var].unique().compute()
+
+                if len(values) > 1:
+                    raise ValueError(
```

Comment:
Maybe useful to test this

Reply:
Is it necessary to loop over tstore_ids? I don't know what is more efficient but maybe 
doing `df[[tstor_tstore_id_vare_id, *static_vars].unique().compute() `and check `len(df) == len(tstore_ids)` is faster ...